ng-deploy/src/builders/actions/__tests__/deploy.spec.ts

```diff
@@ -0,0 +1,78 @@
+import deploy from '../deploy';
+import { FirebaseTools, FirebaseDeployConfig } from '../../../ng-deploy/types';
+import { virtualFs, Path, logging, JsonObject, PathFragment } from '@angular-devkit/core';
+import { BuilderContext, Target, ScheduleOptions, BuilderRun, } from '@angular-devkit/architect/src/index2';
+import { FileBufferLike, HostWatchOptions, HostWatchEvent, FileBuffer } from '@angular-devkit/core/src/virtual-fs/host';
+import { Observable, of } from 'rxjs';
+
+let host: virtualFs.Host;
+let context: BuilderContext
+let firebaseMock: FirebaseTools;
+
+describe('Deploy Angular apps', () => {
+  beforeEach(() => initMocks());
+
+  it('should check if the user is authenticated by invoking list', async () => {
+    const spy = spyOn(firebaseMock, 'list');
```

Comment:
curious if using jest.fn would make it less verbose?
Also with it should be able to do 
` expect(firebaseMock.list).toHaveBeenCalled();`, which would be much easier to read